### DISCUSSION TITLE
Semi-automated correctness proving of arbitrary (!) concurrent programs

### DISCUSSION THREAD

**Discussion by dumblob** [uwu-tech/Kind#240] [General]
I've just discovered [Armada](https://github.com/microsoft/armada ) and I think it's worth watching this 1 year old video https://www.youtube.com/watch?v=CvfFtkAQho4 to get a grasp of what's possible already today :wink:.

Kind has a very different concept of parallelism, but I still think some of the presented ideas are very helpful even for Kind (especially with regards to [memory models](https://research.swtch.com/mm )).

**Comment by MaisaMilena** [comment]
This presenter made me laugh on every new topic. So funny!
